vic/drivers/cesm/cpl_mct/lnd_comp_mct.F90

```diff
@@ -0,0 +1,722 @@
+    CALL mct_aVect_zero(x2l)
+    CALL mct_aVect_init(l2x, rList=seq_flds_l2x_fields, lsize=lsize)
+    CALL mct_aVect_zero(l2x)
+
+    !--- fill the l2x attribute vector with export data
+    CALL lnd_export_mct(l2x)
+
+    !--- fill some scalar export data
+    CALL seq_infodata_PutData(cdata%infodata, &
+         lnd_present=.TRUE., lnd_prognostic=.TRUE., &
+         sno_present=.FALSE., sno_prognostic=.FALSE.)
+    CALL seq_infodata_PutData(infodata, lnd_nx = nx_global, lnd_ny = ny_global)
+
+    !--- set share log unit back to generic one
+    CALL shr_file_setLogLevel(shrloglev)
+
+    !--- deallocate domain arrays
+    DEALLOCATE(gindex)
+    DEALLOCATE(lon_data)
+    DEALLOCATE(lat_data)
+    DEALLOCATE(area_data)
+    DEALLOCATE(mask_data)
+    DEALLOCATE(frac_data)
+
+  END SUBROUTINE lnd_init_mct
+
+  !--------------------------------------------------------------------------
+  !> @brief   vic lnd model run
+  !--------------------------------------------------------------------------
+  SUBROUTINE lnd_run_mct(EClock, cdata, x2l, l2x, cdata_s, x2s, s2x)
+
+    USE shr_file_mod   , ONLY : shr_file_getUnit, shr_file_setLogLevel, &
+                                shr_file_getLogUnit, shr_file_getLogUnit, &
+                                shr_file_setLogUnit
+    USE seq_timemgr_mod, ONLY : seq_timemgr_EClockGetData, seq_timemgr_StopAlarmIsOn, &
+                                seq_timemgr_RestartAlarmIsOn
+
+    IMPLICIT NONE
+
+    !--- INPUT/OUTPUT PARAMETERS:
+    TYPE(ESMF_Clock), INTENT(in)     :: EClock
+    TYPE(seq_cdata),  INTENT(inout)  :: cdata
+    TYPE(mct_aVect),  INTENT(inout)  :: x2l
+    TYPE(mct_aVect),  INTENT(inout)  :: l2x
+    TYPE(seq_cdata),  INTENT(inout)  :: cdata_s
+    TYPE(mct_aVect),  INTENT(inout)  :: x2s
+    TYPE(mct_aVect),  INTENT(inout)  :: s2x
+
+    !--- Local Variables
+    TYPE(seq_infodata_type), POINTER :: infodata ! CESM information from the driver
+    INTEGER(C_INT)                   :: errno
+    CHARACTER(len=*), PARAMETER      :: subname = 'lnd_run_mct'
+
+    !--- set vic log unit
+    CALL shr_file_getLogUnit(shrlogunit)
+    CALL shr_file_setLogUnit(iulog)
+
+    !--- point to cdata
+    CALL seq_cdata_setptrs(cdata, infodata=infodata)
+
+    !--- get time information
+    CALL seq_timemgr_EClockGetData(EClock, &
+         curr_yr=year, &
+         curr_mon=month, &
+         curr_day=day, &
+         curr_tod=seconds)
+
+    !--- update the clock
+    vclock%current_year = year
+    vclock%current_month = month
+    vclock%current_day = day
+    vclock%current_dayseconds = seconds
+    vclock%state_flag = seq_timemgr_RestartAlarmIsOn(EClock)
+    vclock%stop_flag = seq_timemgr_StopAlarmIsOn(EClock)
+
+    !--- get time of next radiation calc for time dependent albedo calc
+    ! Tony - is this for the coupler or for the land model.  We dont need it
+    ! in VIC
+    !CALL seq_infodata_GetData(infodata, nextsw_cday=nextsw_cday)
```

Comment:
Just leave it there and leave it commented out.  This is important IF vic has time dependent (solar angle dependent) albedos.  If it does, then we need to pass this value into vic and it needs to use it.  Basically, nextsw_cday is the julian day of the next atmosphere shortwave radiation calculation.  That "time" should be used to compute the albedos.  In other words, the atm sets this value and passes it to surface components.  Those surface components are supposed to use this time to "place the sun in the sky" for their albedo computations to pass back to the atm model for the radiation calculation.  If there is no time (solar angle) dependence for surface albedos in VIC, nextsw_cday can be ignored.  But I recommend leaving it as is, commented out, as a reminder in case it's needed in the future.